soroban-sdk/src/contract_event.rs

```diff
@@ -24,43 +32,91 @@ use crate::{env::internal, Env, IntoVal, RawVal, Vec};
 /// # #[cfg(feature = "testutils")]
 /// # fn main() {
 /// #     let env = Env::default();
-/// #     let contract_id = FixedBinary::from_array(&env, [0; 32]);
+/// #     let contract_id = BytesN::from_array(&env, [0; 32]);
 /// #     env.register_contract(&contract_id, Contract);
 /// #     f::invoke(&env, &contract_id);
 /// # }
 /// # #[cfg(not(feature = "testutils"))]
 /// # fn main() { }
 /// ```
 #[derive(Clone)]
-pub struct ContractEvent(Env);
+pub struct Events(Env);
 
-impl Debug for ContractEvent {
+impl Debug for Events {
     fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
-        write!(f, "ContractEvent")
+        write!(f, "Events")
+    }
+}
+
+pub trait IntoTopics {
+    fn into_topics(self, env: &Env) -> Object;
```

Comment:
Could you add docs to the type and the into_topics function?
```suggestion
pub trait IntoTopics {
    fn into_topics(self, env: &Env) -> Vec<RawVal>;
```

Reply:
I've adopted the marker trait suggestion which removes need for the "into" function.